### PR TITLE
Dial back a couple of `assert_return()` uses

### PR DESCRIPTION
**Pull request by mrc0mmand**
Since, at least according to `docs/CODING_STYLE.md`, it should be used only for catching programming errors, not runtime ones.

Prompted by #30029 and https://github.com/systemd/systemd/pull/29915. I guess @evverx's idea of being able to do a debug build where `assert_return()` crashes, to weed out "real" issues, is worth pursuing.

### CODE REVIEW

File: src/mount/mount-tool.c [hunk 1]
```diff
@@ -975,15 +975,14 @@ static int stop_mounts(
...
 static int umount_by_device(sd_bus *bus, sd_device *dev) {
-        _cleanup_(sd_device_unrefp) sd_device *d = NULL;
         _cleanup_strv_free_ char **list = NULL;
         const char *v;
         int r, ret = 0;
 
         assert(bus);
         assert(dev);
 
-        if (sd_device_get_property_value(d, "SYSTEMD_MOUNT_WHERE", &v) >= 0)
```
Comment (bluca) [inline src/mount/mount-tool.c:986]: ...this was completely broken, it means it was never taking SYSTEMD_MOUNT_WHERE into account

**Review by bluca — APPROVED** [review]

**Comment by evverx** [comment]
> it should be used only for catching programming errors, not runtime ones

Agreed. Though as far as I can remember there are quite a few places where it's used instead of `if (...) return` to save a few keystrokes probably :-) Once those assert_returns are gone it should be possible to roll out abort() to get closer to things like
```
Nov 15 12:01:23 systemd-resolved[177199]: Assertion 'e->state != SD_EVENT_FINISHED' failed at src/libsystemd/sd-event/sd-event.c:1399, function sd_event_add_time(). Ignoring.
Nov 15 12:01:23 systemd-resolved[177199]: Assertion 'e->state != SD_EVENT_FINISHED' failed at src/libsystemd/sd-event/sd-event.c:1399, function sd_event_add_time(). Ignoring.
Nov 15 12:01:23 systemd-resolved[177199]: Assertion 'e->state != SD_EVENT_FINISHED' failed at src/libsystemd/sd-event/sd-event.c:1399, function sd_event_add_time(). Ignoring.
Nov 15 12:01:23 systemd-resolved[177199]: Assertion 'e->state != SD_EVENT_FINISHED' failed at src/libsystemd/sd-event/sd-event.c:1399, function sd_event_add_time(). Ignoring.
```

File: src/mount/mount-tool.c (hunk 1 above)
Comment (mrc0mmand) [reply src/mount/mount-tool.c:986]: Looks like this is also in v253 and v254 stable branches, as the commit that introduced it is 961d08cadb814dca9be60be55b119fc3a926e3e2.

**Review by keszybz — COMMENTED** [review]
LGTM too.

**Pull request merged**